stdlib/prettyprint/0/prettyprint.rbs

```diff
@@ -0,0 +1,364 @@
+  #
+  # `newline` is used for line breaks. "n" is used if it is not specified.
+  #
+  # The block is used to generate spaces. {|width| ' ' * width} is used if it is
+  # not given.
+  #
+  def initialize: (?untyped output, ?Integer maxwidth, ?String newline) ?{ (Integer) -> untyped } -> void
+
+  # The output object.
+  #
+  # This defaults to '', and should accept the << method
+  attr_reader output: untyped
+
+  # The maximum width of a line, before it is separated in to a newline
+  #
+  # This defaults to 79, and should be a Fixnum
+  attr_reader maxwidth: Integer
+
+  # The value that is appended to +output+ to add a new line.
+  #
+  # This defaults to "\n", and should be String
+  attr_reader newline: String
+
+  # A lambda or Proc, that takes one argument, of a Fixnum, and returns
+  # the corresponding number of spaces.
+  #
+  # By default this is:
+  #   lambda {|n| ' ' * n}
+  attr_reader genspace: untyped
+
+  # The number of spaces to be indented
+  attr_reader indent: Integer
+
+  # The PrettyPrint::GroupQueue of groups in stack to be pretty printed
+  attr_reader group_queue: PrettyPrint::GroupQueue
+
+  # Returns the group most recently added to the stack.
+  #
+  # Contrived example:
+  #     out = ""
+  #     => ""
+  #     q = PrettyPrint.new(out)
+  #     => #<PrettyPrint:0x82f85c0 @output="", @maxwidth=79, @newline="\n", @genspace=#<Proc:0x82f8368@/home/vbatts/.rvm/rubies/ruby-head/lib/ruby/2.0.0/prettyprint.rb:82 (lambda)>, @output_width=0, @buffer_width=0, @buffer=[], @group_stack=[#<PrettyPrint::Group:0x82f8138 @depth=0, @breakables=[], @break=false>], @group_queue=#<PrettyPrint::GroupQueue:0x82fb7c0 @queue=[[#<PrettyPrint::Group:0x82f8138 @depth=0, @breakables=[], @break=false>]]>, @indent=0>
+  #     q.group {
+  #       q.text q.current_group.inspect
+  #       q.text q.newline
+  #       q.group(q.current_group.depth + 1) {
+  #         q.text q.current_group.inspect
+  #         q.text q.newline
+  #         q.group(q.current_group.depth + 1) {
+  #           q.text q.current_group.inspect
+  #           q.text q.newline
+  #           q.group(q.current_group.depth + 1) {
+  #             q.text q.current_group.inspect
+  #             q.text q.newline
+  #           }
+  #         }
+  #       }
+  #     }
+  #     => 284
+  #      puts out
+  #     #<PrettyPrint::Group:0x8354758 @depth=1, @breakables=[], @break=false>
+  #     #<PrettyPrint::Group:0x8354550 @depth=2, @breakables=[], @break=false>
+  #     #<PrettyPrint::Group:0x83541cc @depth=3, @breakables=[], @break=false>
+  #     #<PrettyPrint::Group:0x8347e54 @depth=4, @breakables=[], @break=false>
+  #
+  def current_group: () -> PrettyPrint::Group
+
+  # Breaks the buffer into lines that are shorter than #maxwidth
+  #
+    def break_outmost_groups: () -> untyped
+
+  # This adds `obj` as a text of `width` columns in width.
+  #
+  # If `width` is not specified, obj.length is used.
+  #
+  def text: (untyped obj, ?Integer width) -> untyped
```

Comment:
I'm wondering if we need `untyped` here as the type of `obj`. Is there any reason to make it other than `String`?

Reply:
This is a good point, although the documentation is a little bit obscure for me. It's implicit the use of String.

Reply:
Yeah, I also got confused a bit. 😓 
As far as I know, `obj` is (implicitly) assumed to be a `String`.